browser/ui/sidebar/sidebar_browsertest.cc

```diff
@@ -393,6 +395,50 @@ IN_PROC_BROWSER_TEST_F(SidebarBrowserTestWithPlaylist, Incognito) {
   sidebar_service->RemoveItemAt(0);
 }
 
+class SidebarBrowserTestWithAIChat : public SidebarBrowserTest {
+ public:
+  SidebarBrowserTestWithAIChat() {
+    feature_list_.InitAndEnableFeature(ai_chat::features::kAIChat);
+  }
+  ~SidebarBrowserTestWithAIChat() override = default;
+
+  base::test::ScopedFeatureList feature_list_;
+};
+
+IN_PROC_BROWSER_TEST_F(SidebarBrowserTestWithAIChat, TabSpecificPanel) {
+  // Collect item indexes for test
+  constexpr auto kGlobalItemType = SidebarItem::BuiltInItemType::kBookmarks;
+  constexpr auto kTabSpecificItemType = SidebarItem::BuiltInItemType::kChatUI;
+  auto global_item_index = model()->GetIndexOf(kGlobalItemType);
+  ASSERT_TRUE(global_item_index.has_value());
+  auto tab_specific_item_index = model()->GetIndexOf(kTabSpecificItemType);
+  ASSERT_TRUE(tab_specific_item_index.has_value());
+  // Open 2 more tabs
+  ASSERT_TRUE(ui_test_utils::NavigateToURLWithDisposition(
+      browser(), GURL("brave://newtab/"),
+      WindowOpenDisposition::NEW_FOREGROUND_TAB,
+      ui_test_utils::BROWSER_TEST_WAIT_FOR_LOAD_STOP));
+  ASSERT_TRUE(ui_test_utils::NavigateToURLWithDisposition(
+      browser(), GURL("brave://newtab/"),
+      WindowOpenDisposition::NEW_FOREGROUND_TAB,
+      ui_test_utils::BROWSER_TEST_WAIT_FOR_LOAD_STOP));
+  ASSERT_EQ(tab_model()->GetTabCount(), 3);
+  // Open a "global" panel from Tab 0
+  tab_model()->ActivateTabAt(0);
+  SimulateSidebarItemClickAt(global_item_index.value());
+  // Open a "tab specific" panel from Tab 1
+  tab_model()->ActivateTabAt(1);
+  SimulateSidebarItemClickAt(tab_specific_item_index.value());
+  // Tab Specific panel should be open when Tab 1 is active
```

Comment:
How about adding a code for changing global item's index while tab specific item is active?

Reply:
Do you mean I should test this:
- Change to Tab with global panel open
- Activate a different global panel for that Tab
- Verify that global panel is the new global panel everywhere
- Change to Tab with tab-specific panel open
- Verify that the correct tab-specific panel is open

Reply:
I wanted to check this scenario in the test.

1. Open global panel
2. Change to another tab and open tab specific panel
3. Change global panel's item change by dnd
4. Change to another tab and still global panel is opened even its index is changed

Reply:
It works based on type and not index, so I don't think that could go wrong. But I added a new test to check it anyway. PTAL https://github.com/brave/brave-core/compare/f79f3848107348e9f95c898e094175201503ca0b..6c710d4b6df4c828412ffa0a176540a2c59b1ff8